src/deploy-commands.js

```diff
@@ -0,0 +1,49 @@
+const { REST, Routes } = require("discord.js");
+const fs = require("node:fs");
+const path = require("node:path");
+require("dotenv").config();
+
+const commands = [];
+const foldersPath = path.join(__dirname, "commands");
+const commandFolders = fs.readdirSync(foldersPath);
+const token = process.env.DISCORD_TOKEN;
+const clientId = process.env.DISCORD_CLIENT_ID;
+const guildId = process.env.DISCORD_GUILD_ID;
+
+for (const folder of commandFolders) {
+  const commandsPath = path.join(foldersPath, folder);
+  const commandFiles = fs
+    .readdirSync(commandsPath)
+    .filter((file) => file.endsWith(".js"));
+  for (const file of commandFiles) {
+    const command = require(path.join(commandsPath, file));
+    if ("data" in command && "execute" in command) {
+      commands.push(command.data.toJSON());
+    } else {
+      console.log(
+        `[WARNING] The command at ${filePath} is missing a required "data" or "execute" property.`
+      );
+    }
+  }
+}
+
+const rest = new REST().setToken(token);
+
+(async () => {
+  try {
+    console.log(
+      `Started refreshing ${commands.length} application (/) commands.`
+    );
+
+    const data = await rest.put(
+      Routes.applicationGuildCommands(clientId, guildId),
+      { body: commands }
+    );
+
+    console.log(
+      `Successfully reloaded ${data.length} application (/) commands.`
+    );
+  } catch (error) {
+    console.error(error);
+  }
+})();
```

Comment:
Instead of autoinvoke you could create a new async function then call it or avoid using `await`. Here are the two proposal's examples:

```js
const init = async () => {
  try {
    console.log(
      `Started refreshing ${commands.length} application (/) commands.`
    );

    const data = await rest.put(
      Routes.applicationGuildCommands(clientId, guildId),
      { body: commands }
    );

    console.log(
      `Successfully reloaded ${data.length} application (/) commands.`
    );
  } catch (error) {
    console.error(error);
  }
}

init()
```

Here is the other alternative avoiding to use await:

```js
  try {
    console.log(
      `Started refreshing ${commands.length} application (/) commands.`
    );

    const data = rest.put(
      Routes.applicationGuildCommands(clientId, guildId),
      { body: commands }
    ).then(() => {
      console.log(
        `Successfully reloaded ${data.length} application (/) commands.`
      )
      });
    } catch (error) {
    console.error(error);
  }
  ```